accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: CaaS Platform support
+- Start Date: 2019-09-24
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Initial integration for managing a CaaS Platform cluster from Uyuni/SUSE Manager.
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni/SUSE Manager is the systems management tool and must offer to the end-user the possibility to manage other SUSE products like CaaS Platform.
+Uyuni/SUSE Manager already supports Kubernetes cluster integration. It works with CaaS Platform version 3 (container engine: Docker).
+
+In this regard, this RFC will cover the following aspects:
+
+- Node status reporting
+- Update the current Kubernetes integration of Uyuni/SUSE Manager to work with CaaS Platform version 4 while still supporting CaaS Platform version 3
+- Document how to deploy CaaS Platform using AutoYaST
+
+# Detailed design
+[design]: #detailed-design
+
+## Node status reporting
+
+All the nodes in the cluster can be registered to Uyuni/SUSE Manager using [bootstrapping methods](https://opensource.suse.com/doc-susemanager/suse-manager/client-configuration/registration-overview.html).
```

Comment:
Maybe I am missing context here, are any code changes expected on the Manager side to support this, at all?

Reply:
We are not expecting any major code change but we are not sure yet: what if the Salt registration commands install software that conflicts with `kubectl` and related? Some buffer is required here.

Reply:
@flavio successfully bootstrapped CaaSP v4 nodes with SUSE Manager. Besides that, `kubectl` (and other k8s code) is written in `go` (aka "statically linked binaries"). The odds of conflict are pretty low.

Reply:
> Maybe I am missing context here, are any code changes expected on the Manager side to support this, at all?

Not for the initial iteration ('mvp'). Later I would hope that Manager implements "read-only" clients, to prevent changes of individual nodes.

Reply:
I'm with Klaus here. The initial scope should really be to make sure all the templates and resources needed (e.g. AutoYaST XML, repos) work with SUSE Manager and are available for consumption and that the process is properly documented.
We need that part ASAP (December release) and should then start working on the improvements that require code changes in Uyuni/Manager 4.1 alpha/beta in iterative steps.

Reply:
> We are not expecting any major code change but we are not sure yet: what if the Salt registration commands install software that conflicts with kubectl and related? Some buffer is required here.

I understand that.

My POV is that any RFC should describe an implementation plan with the main goal being to guide the developer who will receive it through implementation. (A second, not-less-important objective is to present explored alternatives with their pros and cons).

So my suggestion in this particular case would be to limit the main message about this section to something like "only testing is needed" - and to reframe that all other (important!) information here is as context.